base/services/dcomlaunch/network.c

```diff
@@ -104,8 +105,14 @@ CookupNodeId(UCHAR * NodeId)
         *NodeMiddle = *NodeMiddle ^ Luid.HighPart;
     }
 
+    /* Get system drive */
+    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, ARRAYSIZE(szDrive) - 1))
```

Comment:
For services, it is more robust and secure to obtain the systemdrive letter not via the environment prompt, but instead by calling the [GetSystemDirectory](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getsystemdirectoryw) function and extracting the drive letter out of it.